src/plot/xy-plot.js

```diff
@@ -187,6 +187,22 @@ class XYPlot extends React.Component {
     }
   };
 
+  _childPropCache = Object.create(null)
```

Comment:
I'm a bit hestiant to add a cache instead of relying on useMemo. How about we wait till the modernization to React 16.8 (#1316) and useMemo instead (after some refactoring of this prop cloning behavior perhaps to using context)?

Alternatively, if you'd like to try out what this looks like via the modernize branch. Feel free to go for it.